File: apps/driver-android/app/src/main/java/ai/bluecollar/towcommand/driver/data/api/TowCommandApi.kt
package ai.bluecollar.towcommand.driver.data.api

import ai.bluecollar.towcommand.driver.data.api.dto.CancelRequest
import ai.bluecollar.towcommand.driver.data.api.dto.DriverProfileDto
import ai.bluecollar.towcommand.driver.data.api.dto.JobDto
import ai.bluecollar.towcommand.driver.data.api.dto.LoginRequest
import ai.bluecollar.towcommand.driver.data.api.dto.LoginResponse
import ai.bluecollar.towcommand.driver.data.api.dto.LogoutRequest
import ai.bluecollar.towcommand.driver.data.api.dto.MeResponse
import ai.bluecollar.towcommand.driver.data.api.dto.MyJobDto
import ai.bluecollar.towcommand.driver.data.api.dto.PhotoUploadRequest
import ai.bluecollar.towcommand.driver.data.api.dto.PhotoUploadResponse
import ai.bluecollar.towcommand.driver.data.api.dto.RefreshRequest
import ai.bluecollar.towcommand.driver.data.api.dto.RefreshResponse
import ai.bluecollar.towcommand.driver.data.api.dto.TransitionRequest
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface TowCommandApi {
    @POST("/auth/login")
    suspend fun login(@Body body: LoginRequest): LoginResponse

    @POST("/auth/refresh")
    suspend fun refresh(@Body body: RefreshRequest): RefreshResponse

    @POST("/auth/logout")
    suspend fun logout(@Body body: LogoutRequest)

    @GET("/auth/me")
    suspend fun me(): MeResponse

    /** Driver-scoped — returns jobs assigned to the driver linked to the current user. */
    @GET("/dispatch/my-jobs")
    suspend fun myJobs(): List<MyJobDto>

    @GET("/dispatch/me/driver")
    suspend fun myDriverProfile(): DriverProfileDto

    @POST("/dispatch/jobs/{id}/transition")
    suspend fun transition(@Path("id") id: String, @Body body: TransitionRequest): JobDto

    @POST("/jobs/{id}/cancel")
    suspend fun cancel(@Path("id") id: String, @Body body: CancelRequest): JobDto

    @POST("/dispatch/jobs/{id}/photos")
    suspend fun uploadJobPhoto(
        @Path("id") id: String,
        @Body body: PhotoUploadRequest,
    ): PhotoUploadResponse
}

File: apps/driver-ios/Packages/Core/Sources/Core/Networking/TowCommandAPI.swift
import Foundation

/// High-level API surface used by the app. Wraps the lower-level `APIClient`.
public protocol TowCommandAPI: Sendable {
    func login(_ body: LoginRequest) async throws -> LoginResponse
    func refresh(_ body: RefreshRequest) async throws -> RefreshResponse
    func logout(_ body: LogoutRequest) async throws
    func me() async throws -> MeResponse
    func myJobs() async throws -> [MyJob]
    func myDriverProfile() async throws -> DriverProfile
    func transition(jobId: String, to: JobStatus, reason: String?) async throws -> Job
    func cancel(jobId: String, reason: String) async throws -> Job
    func uploadJobPhoto(jobId: String, photo: PhotoUploadRequest) async throws -> PhotoUploadResponse
}

public struct TransitionRequest: Codable, Sendable {
    public let to: String
    public let reason: String?
    public init(to: JobStatus, reason: String? = nil) {
        self.to = to.rawValue
        self.reason = reason
    }
}

public struct CancelRequest: Codable, Sendable {
    public let reason: String
    public init(reason: String) { self.reason = reason }
}

public actor LiveTowCommandAPI: TowCommandAPI {
    private let client: APIClient

    public init(client: APIClient) { self.client = client }

    public func login(_ body: LoginRequest) async throws -> LoginResponse {
        try await client.request(.POST, Endpoints.login, body: body, authorize: false)
    }
    public func refresh(_ body: RefreshRequest) async throws -> RefreshResponse {
        try await client.request(.POST, Endpoints.refresh, body: body, authorize: false)
    }
    public func logout(_ body: LogoutRequest) async throws {
        try await client.requestVoid(.POST, Endpoints.logout, body: body, authorize: true)
    }
    public func me() async throws -> MeResponse {
        try await client.request(.GET, Endpoints.me, body: nil, authorize: true)
    }
    public func myJobs() async throws -> [MyJob] {
        try await client.request(.GET, Endpoints.myJobs, body: nil, authorize: true)
    }
    public func myDriverProfile() async throws -> DriverProfile {
        try await client.request(.GET, Endpoints.myDriverProfile, body: nil, authorize: true)
    }
    public func transition(jobId: String, to: JobStatus, reason: String? = nil) async throws -> Job {
        try await client.request(
            .POST,
            Endpoints.transition(jobId: jobId),
            body: TransitionRequest(to: to, reason: reason),
            authorize: true
        )
    }
    public func cancel(jobId: String, reason: String) async throws -> Job {
        try await client.request(
            .POST,
            Endpoints.cancel(jobId: jobId),
            body: CancelRequest(reason: reason),
            authorize: true
        )
    }
    public func uploadJobPhoto(jobId: String, photo: PhotoUploadRequest) async throws -> PhotoUploadResponse {
        try await client.request(
            .POST,
            Endpoints.uploadJobPhoto(jobId: jobId),
            body: photo,
            authorize: true
        )
    }
}
